### DISCUSSION TITLE
[ FF7 + Echo-S ] Crash when loading voice file

### DISCUSSION THREAD

**Discussion by petfriendamy** [julianxhokaxhiu/FFNx#281] [Q&A]
Occasionally, while playing with the Echo-S mod, the game will crash when attempting to load a voice file. This usually seems to happen after playing the game for a while.

Using 7th Heaven v2.3.1.101 with the latest FFNx canary (v1.9.0.198). I've attached the 7th Heaven applog. I also had FFNx generate a crash.dmp, and have included a screenshot from that. If you need the whole file, let me know, I've just left it out for now because the file size is over 1GB.

[applog.txt](https://github.com/julianxhokaxhiu/FFNx/files/7188002/applog.txt)

![crash](https://user-images.githubusercontent.com/65095334/133843093-ded6f2be-d01c-40e6-9870-34ae9068c9c8.png)

**Comment by julianxhokaxhiu** [comment]
I wonder if this is related to #273

What do you think @myst6re ?

**Comment by julianxhokaxhiu** [comment]
@petfriendamy I think you uploaded the wrong log file, because the one in the issue is about 7th. Please upload the `FFNx.log` file. If the size is too big, consider zipping it or using 7-zip and zip it as 7z.

Moved this issue into a discussion as this is related to 7th and not only FFNx on its own.

**Comment by petfriendamy** [comment]
Sorry, all these logs get a bit confusing, haha. Here's the FFNx.log:
[FFNx.log](https://github.com/julianxhokaxhiu/FFNx/files/7190493/FFNx.log)

Actually, the log was pretty empty--I guess I'd run the game again since I made the report--so I did another playthrough just now until it crashed again. Interestingly, the crash happened at around the same time (just after escaping the church with Aerith).

**Reply by julianxhokaxhiu** [reply]
Looks like it crashed when trying to play a voice file. Please enable `trace_voice` and try again. Post the log when it crashes, thanks.

**Reply by petfriendamy** [reply]
Got it:
[FFNx.log](https://github.com/julianxhokaxhiu/FFNx/files/7192543/FFNx.log)

**Reply by julianxhokaxhiu** [reply]
Does the file `F:\Games\FINAL FANTASY VII/voice/chrin_1a/9.ogg` exist on your disk? As this Echo-S is using 7th, you might as well check if 7th is really passing correctly this file to FFNx.

**Reply by petfriendamy** [reply]
There is nothing in F:\Games\FINAL FANTASY VII\voice\ except a config.toml file. Maybe it is a 7H issue, then?

**Comment by petfriendamy** [comment]
Okay, for the sake of objectivity, I copied most of the Echo-S mod files to the Steam version, removing 7H as a factor. Still crashes. (Though it made it further into the game this time. Not sure if that means anything?)

[FFNx.log](https://github.com/julianxhokaxhiu/FFNx/files/7198169/FFNx.log)

**Reply by julianxhokaxhiu** [reply]
Thanks, I'll check the log this evening and let you know. Not having 7th as a factor helps a lot.

**Reply by julianxhokaxhiu** [reply]
Please try the latest Canary and let me know how it goes. It should prevent crashing, but you might not hear some audio files ( voice files in your case ). If that happens, please check the `FFNx.log` file for any ERROR that looks like this: `Cannot load path/to/file.ogg with vgmstream`.

**Reply by petfriendamy** [reply]
Good news, it didn't crash! Instead, it got to the point where it usually crashes and stopped playing voice files altogether.

[FFNx.log](https://github.com/julianxhokaxhiu/FFNx/files/7207553/FFNx.log)

**Reply by julianxhokaxhiu** [reply]
Thank you, I'll try to improve the error message with more context so we can understand what is going on.

**Reply by julianxhokaxhiu** [reply]
Allright, please try with the latest canary, and as soon as it happens again please send the log!

**Reply by petfriendamy** [reply]
Here you go! I tried playing a bit longer this time to see if the voices would come back on the next screen, but they did not.

[FFNx.log](https://github.com/julianxhokaxhiu/FFNx/files/7228182/FFNx.log)
